docs/guides/component-testing.mdx

```diff
@@ -0,0 +1,176 @@
+# Компонентное тестирование (экспериментальное)
+
+## Введение
+
+Узнайте больше об этом в нашем блоге "[Компонентное тестирование][blog-component-testing]".
+
+## Пример
+
+```(typescript)
+import { render } from '@testing-library/react';
+import Component from '../Component';
+
+it('should render react component', async ({browser}) => {
+    // Отрендерить компонент на изолированной странице
+    render(<Component />);
+
+    // Найти кнопку внутри компонента и кликнуть по ней
+    const button = await browser.$("button");
+    await button.click();
+
+    // Проверить, что текст кнопки имеет ожидаемое значение
+    await expect(button).toHaveText("count is 1");
+});
+```
+
+## Как начать?
+
+Давайте настроим тестирование реакт компонентов, написанных на Typescript.
+
+### Шаг 1: Установка Testplane и необходимых зависимостей
+
+```bash
+npm init testplane@latest
+npm i vite @vitejs/plugin-react @testing-library/react --save-dev
+npm i react --save
+```
+
+### Шаг 2: Создание конфигурации Vite и подключение плагина react
+
+```typescript
+// vite.config.ts
+import { defineConfig } from 'vite';
+import react from '@vitejs/plugin-react';
+
+export default defineConfig({
+    plugins: [
+        react(),
+    ]
+});
+```
+
+### Шаг 3: Настройка запуска тестов в браузере (используя опцию testRunEnv)
+
+```typescript
+// .testplane.conf.ts
+export const {
```

Comment:
Please make sure that all code blocks contain an actual valid typescript code. This one is not due to export const without a name.

I've described why here: https://github.com/gemini-testing/testplane-docs/pull/23#discussion_r1711291877